lib/ast-utils.js

```diff
@@ -551,5 +552,37 @@ module.exports = {
             // no default
         }
         return 18;
+    },
+
+    /**
+     * Checks whether a given node is a loop node or not.
+     * The following types are loop nodes:
+     *
+     * - DoWhileStatement
+     * - ForInStatement
+     * - ForOfStatement
+     * - ForStatement
+     * - WhileStatement
+     *
+     * @param {ASTNode|null} node - A node to check.
+     * @returns {boolean} `true` if the node is a loop node.
+     */
+    isLoopNode: function(node) {
+        return Boolean(node && anyLoopPattern.test(node.type));
```

Comment:
oops, I will add tomorrow (JST)

2016年7月11日(月) 22:05 Oleg Gaidarenko <notifications@github.com>:

> In lib/ast-utils.js
> <https://github.com/eslint/eslint/pull/6649#discussion_r70252919>:
>
> > +
> > +    /**
> > +     * Checks whether a given node is a loop node or not.
> > +     * The following types are loop nodes:
> > +     *
> > +     * - DoWhileStatement
> > +     * - ForInStatement
> > +     * - ForOfStatement
> > +     * - ForStatement
> > +     * - WhileStatement
> > +     *
> > +     * @param {ASTNode|null} node - A node to check.
> > +     * @returns {boolean} `true` if the node is a loop node.
> > +     */
> > +    isLoopNode: function(node) {
> > +        return Boolean(node && anyLoopPattern.test(node.type));
>
> Probably worth to add tests for such methods?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/eslint/eslint/pull/6649/files/534b4a63da91a7ee7bb3bced229e269c1a447558..9df950521809d8b4aa8042852fc10edee155554b#r70252919>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB2Rz2_t-vQyN44LhCdM7boewfLk09C7ks5qUj-2gaJpZM4JJSFj>
> .
>